tests/server/test_plugins.py

```diff
@@ -43,79 +137,293 @@ def test_example_config(self):
+            params = dict(self.test_params_default, **{'Command': None})
+            server_plugins.Plugin(name=self.test_plugin_name, params=params)
+        assert 'Command or Module parameter has to be set' in exc_info.value
+
+        with pytest.raises(smoker_exceptions.InvalidConfiguration) as exc_info:
+            test_params = {
+                'Command': 'hostname -f',
+                'Module': 'server_plugins.hostname'
+            }
+            params = dict(self.test_params_default, **test_params)
+            server_plugins.Plugin(name=self.test_plugin_name, params=params)
+        expected = 'Command and Module parameters cannot be set together'
+        assert expected in exc_info.value
+
+        with pytest.raises(smoker_exceptions.InvalidConfiguration) as exc_info:
+            test_params = {
+                'Command': None,
+                'Parser': 'server_plugins.hostnameparser',
+                'Module': 'server_plugins.hostname'
+            }
+            params = dict(self.test_params_default, **test_params)
+            server_plugins.Plugin(name=self.test_plugin_name, params=params)
+        expected = 'Parser can be used only with Command parameter'
+        assert expected in exc_info.value
+
+    def test_schedule_run_with_time(self):
+        next_run = (datetime.datetime.now() + datetime.timedelta(seconds=3))
+        self.plugin.schedule_run(time=next_run)
+        assert self.plugin.next_run == next_run
+
+    def test_schedule_run_with_invalid_time_format(self):
+        with pytest.raises(smoker_exceptions.InvalidArgument) as exc_info:
+            self.plugin.schedule_run(time=15)
+        expected = 'Parameter time has to be an instance of datetime object'
+        assert expected in exc_info.value
+
+        with pytest.raises(smoker_exceptions.InvalidArgument) as exc_info:
+            self.plugin.schedule_run(time=time.ctime())
+        expected = 'Parameter time has to be an instance of datetime object'
+        assert expected in exc_info.value
+
+    def test_schedule_run_now(self):
+        self.plugin.schedule_run(now=True)
+        delta = (datetime.datetime.now() - self.plugin.next_run)
+        assert round(delta.total_seconds(), 0) == 0
+
+    def test_schedule_run_with_interval(self):
+        params = dict(self.test_params_default, **{'Interval': 15})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+
+        delta = (plugin.next_run - datetime.datetime.now())
+        assert round(delta.total_seconds(), 0) == plugin.params['Interval']
+
+    def test_force_to_run_plugin_without_interval_parameter(self):
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=self.test_params_default)
+        plugin.forced = True
+        plugin.run()
+        assert plugin.current_run
+        assert type(plugin.current_run).__name__ == 'PluginWorker'
```

Comment:
There are 2 types : None and 'PluginWorker'
If plugin is running, it should be 'PluginWorker'
If not :  it should be None